benchmarks/brute_force_vs_bvh/brute_force_vs_bvh.cpp

```diff
@@ -0,0 +1,170 @@
+};
+
+template <int DIM>
+void run(int nprimitives, int nqueries, int nrepeats)
+{
+  ExecutionSpace space{};
+  Dummy<DIM> primitives{nprimitives};
+  Dummy<DIM> predicates{nqueries};
+
+  printf("Dimension : %d\n", DIM);
+  printf("Primitives: %d\n", nprimitives);
+  printf("Predicates: %d\n", nqueries);
+  printf("Iterations: %d\n", nrepeats);
+
+  using Box = ArborX::ExperimentalHyperGeometry::Box<DIM>;
+
+  for (int i = 0; i < nrepeats; i++)
+  {
+    unsigned int out_count;
+    {
+      Kokkos::Timer timer;
+      ArborX::BasicBoundingVolumeHierarchy<MemorySpace, Box> bvh{space,
+                                                                 primitives};
+
+      Kokkos::View<int *, ExecutionSpace> indices("indices_ref", 0);
+      Kokkos::View<int *, ExecutionSpace> offset("offset_ref", 0);
+      bvh.query(space, predicates, indices, offset);
+
+      space.fence();
+      double time = timer.seconds();
+      if (i == 0)
+        printf("Collisions: %.5f\n",
+               (float)(indices.extent(0)) / (nprimitives * nqueries));
+      printf("Time BVH: %lf\n", time);
+      out_count = indices.extent(0);
+    }
+
+    {
+      Kokkos::Timer timer;
+      ArborX::BruteForce<MemorySpace, Box> brute{space, primitives};
+
+      Kokkos::View<int *, ExecutionSpace> indices("indices", 0);
+      Kokkos::View<int *, ExecutionSpace> offset("offset", 0);
+      brute.query(space, predicates, indices, offset);
+
+      space.fence();
+      double time = timer.seconds();
+      printf("Time BF: %lf\n", time);
+      ARBORX_ASSERT(out_count == indices.extent(0));
+    }
+  }
+}
+
+int main(int argc, char *argv[])
+{
+  Kokkos::ScopeGuard guard(argc, argv);
+
+  int dim;
+  int nprimitives;
+  int nqueries;
+  int nrepeats;
+  namespace bpo = boost::program_options;
+  bpo::options_description desc("Allowed options");
+  // clang-format off
+  desc.add_options()
+      ( "help", "help message" )
+      ( "dimension", bpo::value<int>(&dim)->default_value(3), "dimension" )
+      ( "iterations", bpo::value<int>(&nrepeats)->default_value(1), "number of iterations" )
+      ( "predicates", bpo::value<int>(&nqueries)->default_value(5), "number of predicates" )
+      ( "primitives", bpo::value<int>(&nprimitives)->default_value(5), "number of primitives" )
+      ;
+  // clang-format on
+  bpo::variables_map vm;
+  bpo::store(bpo::command_line_parser(argc, argv).options(desc).run(), vm);
+  bpo::notify(vm);
+
+  if (vm.count("help") > 0)
+  {
+    std::cout << desc << '\n';
+    return 1;
+  }
+  ARBORX_ASSERT(nprimitives > 0);
+  ARBORX_ASSERT(nqueries > 0);
+
+  switch (dim)
+  {
+  case 3:
+    run<3>(nprimitives, nqueries, nrepeats);
+    break;
+#if KOKKOS_VERSION >= 30700
+  case 2:
+    run<2>(nprimitives, nqueries, nrepeats);
+    break;
+  case 4:
+    run<4>(nprimitives, nqueries, nrepeats);
+    break;
+  case 5:
+    run<5>(nprimitives, nqueries, nrepeats);
+    break;
+  case 6:
+    run<6>(nprimitives, nqueries, nrepeats);
+    break;
```

Comment:
How long does it take to compile?

Reply:
On my Mac, compiling the driver with 6 instantiations takes 20s. 1 instantiation takes 7s.

Reply:
I am curious whether and by how much it speeds up compilation if we explicitly instantiate in multiple source files

Reply:
quick experiment allowed me to compile 3x faster (see aprokop/ArborX#7)

Reply:
I tried it out. Got 12s. So about 2x slower than a single instantiation, and about 40% faster than multiple in a single file.

Reply:
Please elaborate, I am not sure I understand what you did nor what the conclusion you drew

Reply:
I measured the improvements of your suggestion (ETI) on my machine and compared it to the the timings I posted above. I have not drawn any conclusion. I can live with either, as we tradeoff some speedup for the faster compilation vs having a more complicated build and more files.